pkg/backend/httpstate/client/client.go

```diff
@@ -1469,22 +1469,31 @@ func (pc *Client) callCopilot(ctx context.Context, requestBody interface{}) (str
 	}
 	defer resp.Body.Close()
 
+	// Read the body first so we can use it for error reporting if needed
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return "", fmt.Errorf("reading response body: %w", err)
+	}
+
```

Comment:
We don't need to move this here.  It should probably still be after the if below.

Reply:
No you didn't, Mikhail did in the commit earliier.